x-pack/plugins/lens/public/heatmap_visualization/chart_component.tsx

```diff
@@ -0,0 +1,268 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import React, { FC, useEffect, useState } from 'react';
+import { i18n } from '@kbn/i18n';
+import {
+  Chart,
+  ElementClickListener,
+  Heatmap,
+  HeatmapBrushEvent,
+  HeatmapElementEvent,
+  HeatmapSpec,
+  ScaleType,
+  Settings,
+} from '@elastic/charts';
+import { VisualizationContainer } from '../visualization_container';
+import { HeatmapRenderProps } from './types';
+import './index.scss';
+import { LensBrushEvent, LensFilterEvent } from '../types';
+import { desanitizeFilterContext } from '../utils';
+import { search } from '../../../../../src/plugins/data/public';
+
+declare global {
+  interface Window {
+    /**
+     * Flag used to enable debugState on elastic charts
+     */
+    _echDebugStateFlag?: boolean;
+  }
+}
+
+export const HeatmapComponent: FC<HeatmapRenderProps> = ({
+  data,
+  args,
+  timeZone,
+  formatFactory,
+  onClickValue,
+  onSelectRange,
+}) => {
+  const isDarkTheme = false;
```

Comment:
Dark mode should be set dynamically, not statically, by using `chartsThemeService.useDarkMode()`. Otherwise the text and border color is not high contrast in dark mode.

Reply:
Fixed in https://github.com/elastic/kibana/pull/97978/commits/454873151afab867d4efafba116be681abfa8483